### ISSUE TITLE
Why do you use  tgt_soft_label.detach() in Ladv? 

### ISSUE THREAD

**Issue by Sarmadfismael** [JDAI-CV/FADA#28]
thank you for sharing the code, I have a quaestio related with Ladv, why do you use the detach () in Ladv in this line:        https://github.com/JDAI-CV/FADA/blob/a1c6403963184a3427eda68cc94b03ff6143368a/train_adv.py#L186

I ask this question because in training of semantic segmentation we need tgt_soft_label to fool the disc model?
